### ISSUE TITLE
Provide default alt text for code generated images

### ISSUE THREAD

**Issue by rgaiacs** [jupyter/nbconvert#1993]
Consider a cell with

```python
from matplotlib import pyplot as plt

plt.plot([1, 2, 3], [1, 2, 3]);
```

The default rendering of the output is

```
<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>




<div class="output_png output_subarea ">
<img src="data:image/png;base64," >
</div>

</div>

</div>
</div>

</div>
```

Should we have an default `alt` text?

```
<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>




<div class="output_png output_subarea ">
<img src="data:image/png;base64," alt="Image generated using the above code." >
</div>

</div>

</div>
</div>

</div>
```

Related issues and pull request: https://github.com/jupyter/nbconvert/pull/1112

cc @Jenneh @isabela-pf

**Comment by tonyfast** [comment]
I'll agree that adding alt text is meaningful. Not all images need alt, and the existence of `alt` text differentiates between  [decorative images](https://www.w3.org/WAI/tutorials/images/decorative/) and [informative images](https://www.w3.org/WAI/tutorials/images/informative/). When images or figures are used in outputs we can assume they are informative and a lack of alt text is a impending violation.

With that said, coming up with a meaningful default message is completely ill defined. So yes we should have `alt` text, but what that should be is hard to imagine. `alt` should describe the image itself and we need to avoid misusing alt text. We can't make assumptions about the compute context or infer the meaning of the figure. The best we can say is "there is an informative image here, it is not decorative". Ultimately, we are delivering bad news to visually impaired users that they can't participate in all of the information, but at least the existence of the figure is announced because it is not decorative anymore.

another practical implementation would be some guard rails that produce errors (strict more) or warnings that they are producing inaccessible images or content.
